pkg/sql/parser/sql-gen.sh

```diff
@@ -1,23 +1,43 @@
 #!/usr/bin/env bash
 
-# This is used through bazel when generating sql.go. Look at BUILD.bazel for
+# This is used through bazel when generating sql.go and plpgsql.go.
+# Look at BUILD.bazel in pkg/sql/parser or pkg/plpgsql/parser for
 # usage.
 
 set -euo pipefail
 
-awk '/func.*sqlSymUnion/ {print $(NF - 1)}' $1 | \
-    sed -e 's/[]\/$*.^|[]/\\&/g' | \
-    sed -e "s/^/s_(type|token) <(/" | \
-    awk '{print $0")>_\\1 <union> /* <\\2> */_"}' > types_regex.tmp
-
-sed -E -f types_regex.tmp < $1 | \
-    awk -f $2 | \
-    sed -Ee 's,//.*$$,,g;s,/[*]([^*]|[*][^/])*[*]/, ,g;s/ +$$//g' > sql-gen.y
-rm types_regex.tmp
-
-ret=$($4 -p sql -o $3 sql-gen.y); \
-  if expr "$ret" : ".*conflicts" >/dev/null; then \
-    echo "$ret"; exit 1; \
-  fi;
-rm sql-gen.y
-$5 -w $3
+
+if [ "$#" -eq 5 ]; then
```

Comment:
This is a good start. But we can improve from here because as you can see that the most complicated logic of this script is not shared. The trick here is to use variables. The input argument can be assigned to named variables so that we don't need to have the 2 big blocks of mostly duplicate logic. For example:
```sh
GOYACC_BIN=
if [ "$#" -eq 5]; then
    GOYACC_BIN=$4
else
    GOYACC_BIN=$3
fi
```
Then you can use `$GOYACC` in a shared block.

Reply:
Also I think this file can be moved to a share package instead of sitting in `pkg/sql/parser`.